### ISSUE TITLE
Set DisplayName in the Theory and append (or overwrite) in InlineData

### ISSUE THREAD

**Issue by Jayman1305** [xunit/xunit#682]
Is there any  way i can modify the Display Name in the Inline data to make my test explorer more logical. 
So basically im looking at something like this.
[Theory(DisplayName = "Installed Browser")]
[InlineData(DisplayName = " - Chrome", WebDriver.ChromeDriver, @"C:\Users\James\Documents\ScreenShots\Chrome",30)]

so in my test explorer i can see the test As Installed Browser - Chrome

It would alos be fine if there was any way to override the DisplayName in the inline data itself. 

Thanks

**Comment by yannduran** [comment]
I also was looking to do the same thing basically. I think it's a very good suggestion!

**Comment by joshuaduffy** [comment]
You can use an `enum` in inline data. The way we do this is by using a `switch` statement that sets the driver sets the driver depending on what value was passed in.

**Comment by yannduran** [comment]
@joshuaduffy How does an enum help with the test's DisplayName? I was quite surprised to find that the Theory attribute didn't have a DisplayName parameter like the Fact attribute does. It seems like a very common requirement to want to tidy up the names displayed in Test Explorer.

In Test Explorer, the tests created by a Theory have really ugly (and often very long) names.

**Comment by bradwilson** [comment]
@yannduran wrote:

> I was quite surprised to find that the Theory attribute didn't have a DisplayName parameter like the Fact attribute does.

`TheoryAttribute` does contain `DisplayName`, because it inherits it from `FactAttribute`.

**Comment by yannduran** [comment]
@bradwilson That was actually the first thing that I tried, lol, but I remember getting an error message saying something about Theory not having a constructor that takes one parameter.

However, I just tried it again, and I can compile ok, but when the tests are output to Test Manager the display name that I passed into the Theory attribute is simply ignored, and the ugly, long default names are displayed instead.

I appreciate the help from both you and @joshuaduffy, so thanks! But the problem remains.
